### PR TITLE
Fix store_variable_comment route

### PR DESCRIPTION
**Pull request by zaucker**
Adapt and cleanup frontend/backend interface.

### CODE REVIEW

File: lib/Agrammon/DB/Dataset.pm6 [hunk 1]
```diff
@@ -217,16 +217,14 @@ class Agrammon::DB::Dataset does Agrammon::DB {
         }
     }
 
-    method store-input-comment($var-name, $comment) {
+    method store-input-comment(:$variable, :$comment) {
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:220]: Named parameters are optional by default; is that what you wanted?

File: lib/Agrammon/Web/Routes.pm6 [hunk 2]
```diff
@@ -282,9 +282,12 @@ sub application-routes(Agrammon::Web::Service $ws) {
 
         # test
         post -> LoggedIn $user, 'store_variable_comment' {
-            request-body -> (:$dataset, :$variable, :$comment) {
-                my $data = $ws.store-input-comment($user, $dataset, $variable, $comment);
-                content 'application/json', $data;
+            request-body -> %data {
+                my $ret = $ws.store-input-comment(
+                    $user,
+                    :dataset(%data<dataset>), :variable(%data<variable>), :comment(%data<comment>)
```
Comment (jnthn) [inline lib/Agrammon/Web/Routes.pm6:288]: Probably also better unpacked in the `request-body` line.

File: lib/Agrammon/DB/Dataset.pm6 (hunk 1 above)
Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6:220]: Oh, `$variable` should not be optional. `$comment` should be optional to be able to "clear" the comment.

Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6:277]: Fixed.

File: lib/Agrammon/Web/Routes.pm6 (hunk 2 above)
Comment (zaucker) [reply lib/Agrammon/Web/Routes.pm6]: Done

**Comment by zaucker** [comment]
Also changed to fit with the other OpenAPI routes.

**Review by jnthn — APPROVED** [review]

**Pull request merged**